auslib/web/public/base.py

```diff
@@ -61,10 +61,12 @@ def apply_security_headers(response):
 
 @app.errorhandler(404)
 def fourohfour(error):
-    """We don't return 404s in AUS. Instead, we return empty XML files"""
-    response = make_response('<?xml version="1.0"?>\n<updates>\n</updates>')
-    response.mimetype = 'text/xml'
-    return response
+    if re.match("^/update", request.path):
+        """We don't return 404s in AUS. Instead, we return empty XML files"""
+        response = make_response('<?xml version="1.0"?>\n<updates>\n</updates>')
+        response.mimetype = 'text/xml'
+        return response
+    return Response(status=404, mimetype="text/plain")
```

Comment:
The error handle expects an exception, see the flow:
https://github.com/pallets/flask/blob/c74f46979a8b8358437bd7f76e478d04248a9c72/flask/app.py#L1727
https://github.com/pallets/flask/blob/c74f46979a8b8358437bd7f76e478d04248a9c72/flask/app.py#L1606
https://github.com/pallets/flask/blob/c74f46979a8b8358437bd7f76e478d04248a9c72/flask/app.py#L1638
https://github.com/pallets/flask/blob/c74f46979a8b8358437bd7f76e478d04248a9c72/flask/app.py#L1561